x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -264,9 +267,44 @@ protected Query asQuery(BinaryComparison bc, TranslatorHandler handler) {
 
         public static void checkBinaryComparison(BinaryComparison bc) {
             Check.isTrue(bc.right().foldable(),
-                         "Line {}:{}: Comparisons against fields are not (currently) supported; offender [{}] in [{}]",
-                         bc.right().sourceLocation().getLineNumber(), bc.right().sourceLocation().getColumnNumber(),
-                         Expressions.name(bc.right()), bc.symbol());
+                "Line {}:{}: Comparisons against fields are not (currently) supported; offender [{}] in [{}]",
+                bc.right().sourceLocation().getLineNumber(), bc.right().sourceLocation().getColumnNumber(),
+                Expressions.name(bc.right()), bc.symbol());
+            bc.left().forEachDown(ArithmeticOperation.class, op -> checkFieldsUsageInArithmeticOperation(op, bc));
+        }
+
+        @SuppressWarnings("rawtypes")
+        public static void checkFieldsUsageInArithmeticOperation(ArithmeticOperation a, BinaryPredicate bc) {
+            Set<FieldAttribute> leftFields = collectFields(a.left(), bc);
+            Set<FieldAttribute> rightFields = collectFields(a.right(), bc);
+            boolean containsAll = (leftFields.size() > rightFields.size() && leftFields.containsAll(rightFields))
+                || (leftFields.size() < rightFields.size() && rightFields.containsAll(leftFields))
+                || leftFields.containsAll(rightFields);
```

Comment:
Nit: couldn't the last condition be done away with by adding the equality to the the first condition?

Reply:
Indeed, nice catch. Will change.